### PR TITLE
Various Coverity fixups

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
Just a few coverity fixups, in test code mainly, to appease it.

##### Checklist

- [ ] documentation is added or updated
- [X] tests are added or updated

### CODE REVIEW

**Comment by FdaSilvaYY** [comment]
Can be backported to 1.1.0 too.

**Comment by mattcaswell** [comment]
> Can be backported to 1.1.0 too.

1.1.1 would be ok. 1.1.0 is in security fix only mode.

**Comment by FdaSilvaYY** [comment]
> 
> 
> > Can be backported to 1.1.0 too.
> 
> 1.1.1 would be ok. 1.1.0 is in security fix only mode.

I"ve checked against 1.1.1 , and confuse its will writing.

File: test/evp_test.c [hunk 1]
```diff
@@ -1811,9 +1811,10 @@ static int encode_test_run(EVP_TEST *t)
...
         EVP_ENCODE_CTX *encode_ctx;
 
-        if (!TEST_ptr(encode_ctx = EVP_ENCODE_CTX_new())
-                || !TEST_ptr(encode_out =
-                        OPENSSL_malloc(EVP_ENCODE_LENGTH(expected->input_len))))
+        if (!TEST_ptr(encode_out =
+                      OPENSSL_malloc(EVP_ENCODE_LENGTH(expected->input_len))))
+            goto err;
+        if (!TEST_ptr(encode_ctx = EVP_ENCODE_CTX_new()))
             goto err;
```
Comment (mspncp) [inline test/evp_test.c]: Although swapping the order order of the tests formally solves the leak, it appears to me a little bit over-sophisticated and requires a second thought. I would prefer when `encode_ctx` and `decode_ctx` would be treated analogously, i.e. they should both be declared and NULL-initialized at function scope  (at line 1804)

https://github.com/openssl/openssl/blob/4c66c66c9e2497633102515ce30e217a9f13ed55/test/evp_test.c#L1804

and cleaned up at the `err:` label.

https://github.com/openssl/openssl/blob/4c66c66c9e2497633102515ce30e217a9f13ed55/test/evp_test.c#L1866

File: test/evp_test.c [hunk 2]
```diff
@@ -1818,8 +1818,11 @@ static int encode_test_run(EVP_TEST *t)
...
 
         EVP_EncodeInit(encode_ctx);
-        EVP_EncodeUpdate(encode_ctx, encode_out, &chunk_len,
-                         expected->input, expected->input_len);
+        if (!TEST_true(EVP_EncodeUpdate(encode_ctx, encode_out, &chunk_len,
+                                        expected->input, expected->input_len))){
+            EVP_ENCODE_CTX_free(encode_ctx);
```
Comment (mspncp) [inline test/evp_test.c]: `goto err;` only, and cleanup `encode_ctx` below, see my other comment.

File: test/evp_test.c (hunk 1 above)
Comment (mspncp) [reply test/evp_test.c]: In particular, this hunk can be reverted if `decode_ctx` is declared at function scope.

Comment (mspncp) [reply test/evp_test.c]: Maybe it also makes sense to join 4c66c66c9e2 and f27d4e763d8 into a single commit?

Comment (FdaSilvaYY) [reply test/evp_test.c]: > 
> Maybe it also makes sense to join [4c66c66](https://github.com/openssl/openssl/commit/4c66c66c9e2497633102515ce30e217a9f13ed55) and [f27d4e7](https://github.com/openssl/openssl/commit/f27d4e763d8baf2fd22d1523dda8321a01440b57) into a single commit?

these commit are not fixing the same type of Coverity diag message, that's why I don't squash them.

File: test/evp_test.c (hunk 2 above)
Comment (FdaSilvaYY) [reply test/evp_test.c]: ok, let's do it.

File: test/evp_test.c (hunk 1 above)
Comment (FdaSilvaYY) [reply test/evp_test.c]: I squash the 2 "  Fix CID 13xxxx: Resource leak in test/evp_test.c " in one commit, and reorder them a bit .

**Comment by FdaSilvaYY** [comment]
Fixed and simplify my fixup, needs re-approval.

File: test/evp_test.c [hunk 3]
```diff
@@ -1798,6 +1801,7 @@ static int encode_test_run(EVP_TEST *t)
     ENCODE_DATA *expected = t->data;
     unsigned char *encode_out = NULL, *decode_out = NULL;
     int output_len, chunk_len;
+    EVP_ENCODE_CTX *encode_ctx = NULL;
     EVP_ENCODE_CTX *decode_ctx;
```
Comment (mspncp) [inline test/evp_test.c]: Nit:  I know the NULL initialization is not necessary for `decode_ctx`. But I have an obsessive addiction to consistency: Why not join the two lines as

```C
EVP_ENCODE_CTX *encode_ctx = NULL, *decode_ctx = NULL;
```` 
to make it consistent with L1802?

https://github.com/openssl/openssl/blob/e89d6956eef73333f4ed6b6858e948a8cda40e96/test/evp_test.c#L1802

But if you don't share my obsession, you are free to ignore this comment ;-)

File: test/evp_test.c [hunk 4]
```diff
@@ -1817,8 +1817,10 @@ static int encode_test_run(EVP_TEST *t)
             goto err;
 
         EVP_EncodeInit(encode_ctx);
-        EVP_EncodeUpdate(encode_ctx, encode_out, &chunk_len,
-                         expected->input, expected->input_len);
+        if (!TEST_true(EVP_EncodeUpdate(encode_ctx, encode_out, &chunk_len,
+                                        expected->input, expected->input_len))){
+            goto err;
+        }
```
Comment (mspncp) [inline test/evp_test.c]: Coding style nit: braces not necessary. Personally, I wouldn't complain if you prefer to leave them in place, but then you would need to insert a space before the opening brace in line 1821: `){` -> `) {`

File: test/evp_test.c [hunk 5]
```diff
@@ -1798,31 +1801,30 @@ static int encode_test_run(EVP_TEST *t)
     ENCODE_DATA *expected = t->data;
     unsigned char *encode_out = NULL, *decode_out = NULL;
     int output_len, chunk_len;
-    EVP_ENCODE_CTX *decode_ctx;
+    EVP_ENCODE_CTX *decode_ctx = NULL, encode_ctx = NULL;
```
Comment (mattcaswell) [inline test/evp_test.c]: Oops. Travis is barfing on this line. I think this should be `*encode_ctx`.

Comment (mspncp) [reply test/evp_test.c]: Yes, that's what I suggested. Sorry for overlooking it in the fixup. :-/

**Comment by FdaSilvaYY** [comment]
Oups, sorry for the typo.
Fully fixed, CI's are green.

**Comment by paulidale** [comment]
Merged to master and 1.1.1.  Thanks all.

**Comment by paulidale** [comment]
I shouldn't have squashed this, apologies.

**Pull request closed without merging**